### PR TITLE
Factory reset

### PR DESCRIPTION
**Pull request by sosthene-nitrokey**
This PR adds full device factory reset functionality

Depends on

- https://github.com/Nitrokey/admin-app/pull/15
- https://github.com/trussed-dev/littlefs2/pull/47
- https://github.com/Nitrokey/opcard-rs/pull/188

### CODE REVIEW

**Comment by sosthene-nitrokey** [comment]
Relevant nitropy PR: https://github.com/Nitrokey/pynitrokey/pull/472

**Comment by daringer** [comment]
can you please rebase on top of main to make the tests work? + a `cargo fmt` might be missing?

File: components/apps/src/dispatch.rs [hunk 1]
```diff
@@ -236,8 +270,8 @@ pub enum Backend {
     Auth,
     #[cfg(feature = "backend-rsa")]
     SoftwareRsa,
-    #[cfg(feature = "backend-staging")]
     Staging,
+    StagingManage,
```
Comment (robin-nitrokey) [inline components/apps/src/dispatch.rs:274]: Please add a comment explaining why we have this separate ID.  Might not be obvious in the future.

File: components/apps/src/dispatch.rs [hunk 2]
```diff
@@ -300,30 +331,48 @@ impl<T: Twi, D: Delay> ExtensionId<AuthExtension> for Dispatch<T, D> {
...
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn file_preserve() {
+        assert!(should_preserve_file(path!("/fido/sec/00")));
+        assert!(should_preserve_file(path!("/fido/x5c/00")));
+        assert!(should_preserve_file(path!("/fido/sec/01")));
+        assert!(should_preserve_file(path!("/fido/x5c/01")));
```
Comment (robin-nitrokey) [inline components/apps/src/dispatch.rs:375]: please also add tests for the Trussed keys
https://github.com/Nitrokey/nitrokey-3-firmware/blob/056b62c1369d956ffafdeecba0dec81f5324ed55/components/provisioner-app/src/lib.rs#L95-L103

**Review by robin-nitrokey — APPROVED** [review]

**Comment by tlaurion** [comment]
Let me know what rom to test, I have a nk3 that would require full device factory reset. Otherwise HOTP cannot happen I I cannot reset secrets with nitropy.

**Comment by sosthene-nitrokey** [comment]
@tlaurion You should be able to do  that with the next `test` release.

**Pull request merged**